### DISCUSSION TITLE
NATS Pub-Sub with storing messages in NATS when there is no subscribers?

### DISCUSSION THREAD

**Discussion by dmitryrn** [nats-io/nats-server#3330] [Q&A] [answered]
I know there is JetStream, but I'm wondering, can you use NATS' Pub-Sub and guarantee to not lose messages when all subscriber services are down and someone is publishing a message? Presumable storing there messages in NATS cluster in RAM.

**Comment by ripienaar** [comment, marked as answer]
That's the problem JetStream solves, NATS Core is best efforts.

**Reply by ripienaar** [reply]
Best efforts mean if no-one is listening the message is lost, if the network is interrupted, the message is lost.  There is no persistence, retries etc in the core broker.

Clients can use request-reply to know if the message was received and do own retry.

Otherwise stick them in a stream and consume from the stream
